### PR TITLE
Also allow drag and drop of vector layers to non-GPKG OGR database datasets

### PR DESCRIPTION
**Pull request by nyalldawson**
For instance allows drag and drop of a layer onto a GDB to import that dataset into the geodatabase

### CODE REVIEW

File: src/app/browser/qgsinbuiltdataitemproviders.cpp [hunk 1]
```diff
@@ -1600,6 +1602,145 @@ void QgsDatabaseItemGuiProvider::populateContextMenu( QgsDataItem *item, QMenu *
...
+            exists = true;
+          }
+        }
+
+        if ( ! exists || QMessageBox::question( nullptr, tr( "Overwrite Layer" ),
+                                                tr( "Destination layer <b>%1</b> already exists. Do you want to overwrite it?" ).arg( dropUri.name ), QMessageBox::Yes |  QMessageBox::No ) == QMessageBox::Yes )
+        {
+          QgsVectorLayer *vectorSrcLayer = qobject_cast < QgsVectorLayer * >( srcLayer );
+          QVariantMap options;
+          //  options.insert( QStringLiteral( "driverName" ), QStringLiteral( "GPKG" ) );
```
Comment (nirvn) [inline src/app/browser/qgsinbuiltdataitemproviders.cpp:1695]: Should this be removed?

File: src/app/browser/qgsinbuiltdataitemproviders.cpp [hunk 2]
```diff
@@ -1600,6 +1602,145 @@ void QgsDatabaseItemGuiProvider::populateContextMenu( QgsDataItem *item, QMenu *
...
+          {
+            if ( error != Qgis::VectorExportResult::UserCanceled )
+            {
+              QgsMessageOutput *output = QgsMessageOutput::createMessageOutput();
+              output->setTitle( tr( "Import to database" ) );
+              output->setMessage( tr( "Failed to import some vector layers!\n\n" ) + errorMessage, QgsMessageOutput::MessageText );
+              output->showMessage();
+            }
+          } );
+        } // do not overwrite
```
Comment (nirvn) [inline src/app/browser/qgsinbuiltdataitemproviders.cpp:1720]: Kill those end of conditional block comments? I prefer the ones (that exists) at the beginning of the conditions :)

File: src/core/browser/qgsfilebaseddataitemprovider.cpp [hunk 3]
```diff
@@ -248,17 +248,22 @@ QgsAbstractDatabaseProviderConnection *QgsFileDataCollectionItem::databaseConnec
   // do not print errors, but write to debug
   CPLPushErrorHandler( CPLQuietErrorHandler );
   CPLErrorReset();
-  gdal::dataset_unique_ptr hDS( GDALOpenEx( path().toUtf8().constData(), GDAL_OF_VECTOR | GDAL_OF_READONLY, nullptr, nullptr, nullptr ) );
+  GDALDriverH hDriver = GDALIdentifyDriverEx( path().toUtf8().constData(), GDAL_OF_VECTOR, nullptr, nullptr );
```
Comment (nirvn) [inline src/core/browser/qgsfilebaseddataitemprovider.cpp:251]: Nice, TIL.

**Review by nirvn — APPROVED** [review]

**Comment by agiudiceandrea** [comment]
Hi @nyalldawson, it seems this functionality has an issue in some circumstances. Please see https://github.com/qgis/QGIS/issues/51435.

**Pull request merged**